test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/internalconfig/InternalConfigTest.java

```diff
@@ -0,0 +1,86 @@
+ */
+
+package software.amazon.awssdk.services.internalconfig;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.anyUrl;
+import static com.github.tomakehurst.wiremock.client.WireMock.containing;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.verify;
+
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import java.net.URI;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.protocolrestjsonwithinternalconfig.ProtocolRestJsonWithInternalConfigAsyncClient;
+import software.amazon.awssdk.services.protocolrestjsonwithinternalconfig.ProtocolRestJsonWithInternalConfigClient;
+import software.amazon.awssdk.utils.builder.SdkBuilder;
+
+public class InternalConfigTest {
+
+    @Rule
+    public WireMockRule wireMock = new WireMockRule(0);
+
+    private ProtocolRestJsonWithInternalConfigClient client;
+
+    private ProtocolRestJsonWithInternalConfigAsyncClient asyncClient;
+
+    @Before
+    public void setupClient() {
+        client = ProtocolRestJsonWithInternalConfigClient.builder()
+                                                         .credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create("akid", "skid")))
+                                                         .region(Region.US_EAST_1)
+                                                         .endpointOverride(URI.create("http://localhost:" + wireMock.port()))
+                                                         .build();
+
+        asyncClient = ProtocolRestJsonWithInternalConfigAsyncClient.builder()
+                                                                   .credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create("akid", "skid")))
+                                                                   .region(Region.US_EAST_1)
+                                                                   .endpointOverride(URI.create("http://localhost:" + wireMock.port()))
+                                                                   .build();
+    }
+
+    @Test
+    public void sync_shouldContainInternalUserAgent() {
+        stubResponse();
+        client.allTypes(SdkBuilder::build);
+        verifyUserAgent();
+    }
+
+
+    @Test
+    public void async_shouldContainInternalUserAgent() {
+        stubResponse();
+        asyncClient.allTypes(SdkBuilder::build).join();
+        verifyUserAgent();
+    }
+
```

Comment:
Can you add a negative case, too, for when the client doesn't contain the internal user agent?

Reply:
Sure